plugins/defaultplugins/l2plugin/bd_config.go

```diff
@@ -319,12 +319,29 @@ func (plugin *BDConfigurator) ResolveCreatedInterface(interfaceName string, inte
 	return nil
 }
 
-// ResolveDeletedInterface does nothing
-func (plugin *BDConfigurator) ResolveDeletedInterface(interfaceName string) {
+// ResolveDeletedInterface removes interface from bridge domain operational status
+func (plugin *BDConfigurator) ResolveDeletedInterface(interfaceName string) error {
 	log.DefaultLogger().Print("Interface was removed. Unregister from real state ", interfaceName)
+	// Lookup IfToBdIndexes in order to find a bridge domain for this interface
+	_, meta, found := plugin.IfToBdIndexes.LookupIdx(interfaceName)
+	if !found {
+		log.DefaultLogger().Debugf("Removed interface %s does not belong to any bridge domain", interfaceName)
+		return nil
+	}
+	bdID := meta.(*BridgeDomainMeta).BridgeDomainIndex
+	// Find bridge domain name
+	bdName, meta, found := plugin.BdIndexes.LookupName(bdID)
+	if !found {
+		return fmt.Errorf("unknown bridge domain ID %v", bdID)
+	}
+	err := plugin.LookupBridgeDomainDetails(bdID, bdName)
```

Comment:
What is the intent of having this call here?

err := plugin.LookupBridgeDomainDetails(bdID, bdName)

Please either remove unnecessary call or document the intent in place.

Reply:
this call is updating bridge domain state according to the data in vpp, so technically this is the thing which is currently missing. Maybe the naming could be better but this is the convention taken from other default plugins